client/src/pages/PaymentPortal.js

```diff
@@ -28,20 +32,31 @@ class PaymentPortal extends Component {
   componentDidMount(){
     //TODO: Figure out how it is being passed later : this.props.houseHoldID
     //TODO: Add logic for retrieving householdID
-    //TODO: Replact 5c8680ffad46ec4f26e7b46f with householdID from redux
-    let urlToFetch = API_STUDENT_PROD+ '5c8680ffad46ec4f26e7b46f';
+
+
+    const householdId = this.props.auth.household.householdId;
+    const idFromJwt = jwt_decode(localStorage.getItem('jwtToken'));
+    
+    this.setState({
+      householdId: householdId
+    })
+
+    if (idFromJwt.householdId == this.state.householdId) {
+      let urlToFetch = API_STUDENT_PROD + householdId;
+    
     if (ReactIsInDevelomentMode()){
       //Fetch the student list
-      urlToFetch = API_STUDENT_DEV + '5c8680ffad46ec4f26e7b46f';
-
+      urlToFetch = API_STUDENT_DEV + householdId;
     }
     fetch(urlToFetch)
     .then(response => {if (response.status !== 404) {
+      // TODO: Fix this, it'll break our shit in the future.
```

Comment:
Does this work now or are we still going to end up taking down the site with this inclusion?

Reply:
I think yesterday when we checked this we got a 500 error and it broke the site

Reply:
Hmm wait so that was because of the houseHold ID we're using and what the backend expects. That is something we need to figure out in a meeting, because the backend expects an ID like "5asdfadx23..." but we're using plainText for logging in. So one of them will probably need to be changed.